ntbs-service/DataMigration/NotificationImportService.cs

```diff
@@ -0,0 +1,126 @@
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+using EFAuditer;
+using ntbs_service.DataAccess;
+using ntbs_service.Models;
+using ntbs_service.Models.Enums;
+using Serilog;
+
+namespace ntbs_service.DataMigration
+{
+
+    public interface INotificationImportService
+    {
+        Task<IList<Notification>> ImportNotificationsAsync(string notificationId);
+    }
+
+    public class NotificationImportService : INotificationImportService
+    {
+        private readonly INotificationSearcher notificationSearcher;
+        private readonly INotificationRepository notificationRepository;
+        private readonly NtbsContext _context;
+
+        public NotificationImportService(INotificationSearcher notificationSearcher,
+                             INotificationRepository notificationRepository,
+                             NtbsContext context)
+        {
+            this.notificationSearcher = notificationSearcher;
+            this.notificationRepository = notificationRepository;
+            _context = context;
+        }
+
+        public async Task<IList<Notification>> ImportNotificationsAsync(string notificationId)
+        {
+            LogInformation($"Request to import Notification with Id={notificationId}");
+
+            if (notificationRepository.NotificationWithLtbrOrEtsIdExists(notificationId)) 
+            {
+                LogInformation($"Notification with Id={notificationId} already exists in database");
+                return null;
+            }
+
+            var notificationTask = notificationSearcher.Search(notificationId);
+            Task.WaitAll(notificationTask);
+            var notifications = notificationTask.Result.ToList();

```

Comment:
Is there a reason this wasn't just `var notifications = await notificationSearch.Search(notificationId);`?